Sources/SwiftParser/TokenSpec.swift

```diff
@@ -169,29 +191,50 @@ public struct TokenSpec {
   /// modification of test cases. This should never be used in the parser itself.
   @_spi(AlternateTokenIntrospection)
   public var synthesizedTokenKind: TokenKind {
-    switch rawTokenKind {
-    case .binaryOperator: return .binaryOperator("+")
-    case .dollarIdentifier: return .dollarIdentifier("$0")
-    case .floatLiteral: return .floatLiteral("1.0")
-    case .identifier: return .identifier("myIdent")
-    case .integerLiteral: return .integerLiteral("1")
-    case .keyword: return .keyword(keyword!)
-    case .postfixOperator: return .postfixOperator("++")
-    case .prefixOperator: return .prefixOperator("!")
-    case .rawStringPoundDelimiter: return .rawStringPoundDelimiter("#")
-    case .regexLiteralPattern: return .regexLiteralPattern(".*")
-    case .regexPoundDelimiter: return .regexPoundDelimiter("#")
-    case .stringSegment: return .stringSegment("abc")
-    default: return TokenKind.fromRaw(kind: rawTokenKind, text: "")
+    switch matcher {
+    case .fixedText(let text):
+      return .identifier(String(syntaxText: text))
+    case .keyword(let keyword):
+      return .keyword(keyword)
+    case .tokenKind(let kind):
+      switch kind {
+      case .binaryOperator: return .binaryOperator("+")
+      case .dollarIdentifier: return .dollarIdentifier("$0")
+      case .floatLiteral: return .floatLiteral("1.0")
+      case .identifier: return .identifier("myIdent")
+      case .integerLiteral: return .integerLiteral("1")
+      case .postfixOperator: return .postfixOperator("++")
+      case .prefixOperator: return .prefixOperator("!")
+      case .rawStringPoundDelimiter: return .rawStringPoundDelimiter("#")
+      case .regexLiteralPattern: return .regexLiteralPattern(".*")
+      case .regexPoundDelimiter: return .regexPoundDelimiter("#")
+      case .stringSegment: return .stringSegment("abc")
+      default: return TokenKind.fromRaw(kind: kind, text: "")
+      }
     }
   }
+
+  static func keyword(_ keyword: Keyword) -> TokenSpec {
+    return TokenSpec(keyword)
+  }
+
+  static func fixedText(_ text: SyntaxText) -> TokenSpec {
```

Comment:
I think it would be good to add a short comment that explains when somebody should use `fixedText` vs adding a keyword. The chance of misuse seems quite high to me.